pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -354,51 +385,55 @@ func (n *NetworkPolicyController) createAppliedToGroup(npNsName string, pSel, nS
 	return appliedToGroupUID
 }
 
-// labelsMatchGroupSelector matches a Pod's labels to the
+// labelsMatchGroupSelector matches a ExternalEntity or Pod's labels to the
 // GroupSelector object and returns true, if and only if the labels
 // match any of the selector criteria present in the GroupSelector.
-func (n *NetworkPolicyController) labelsMatchGroupSelector(pod *v1.Pod, podNS *v1.Namespace, sel *antreatypes.GroupSelector) bool {
+func (n *NetworkPolicyController) labelsMatchGroupSelector(obj metav1.Object, ns *v1.Namespace, sel *antreatypes.GroupSelector) bool {
+	objSelector := sel.PodSelector
+	if _, ok := obj.(*v1alpha1.ExternalEntity); ok {
+		objSelector = sel.ExternalEntitySelector
+	}
 	if sel.Namespace != "" {
-		if sel.Namespace != pod.Namespace {
-			// Pods must be matched within the same Namespace.
+		if sel.Namespace != obj.GetNamespace() {
+			// Pods or ExternalEntities must be matched within the same Namespace.
 			return false
 		}
-		if !sel.PodSelector.Matches(labels.Set(pod.Labels)) {
-			// podSelector does not match the Pod's labels.
-			return false
+		if objSelector != nil && objSelector.Matches(labels.Set(obj.GetLabels())) {
+			// podSelector or externalEntitySelector matches the ExternalEntity or Pod's labels.
+			return true
 		}
-		// podSelector matches the Pod's labels.
-		return true
-	} else if sel.NamespaceSelector != nil && sel.PodSelector != nil {
-		// Pod event may arrive before Pod's Namespace event. In this case, we must
-		// ensure that the Pod Namespace is not nil.
-		if podNS == nil || !sel.NamespaceSelector.Matches(labels.Set(podNS.Labels)) {
+		// selector does not match the ExternalEntity or Pod's labels.
+		return false
+	} else if sel.NamespaceSelector != nil && objSelector != nil {
+		// Pod or ExternalEntity event may arrive before Namespace event. In this case, we must
+		// ensure that the Namespace is not nil.
+		if ns == nil || !sel.NamespaceSelector.Matches(labels.Set(ns.Labels)) {
 			// Pod's Namespace do not match namespaceSelector.
 			return false
 		}
-		if !sel.PodSelector.Matches(labels.Set(pod.Labels)) {
-			// Pod's Namespace matches namespaceSelector but Pod's labels do not match
-			// the podSelector.
+		if !objSelector.Matches(labels.Set(obj.GetLabels())) {
+			// ExternalEntity or Pod's Namespace matches namespaceSelector but
+			// labels do not match the podSelector or externalEntitySelector.
 			return false
 		}
-		// Pod's Namespace matches namespaceSelector and Pod's labels matches
-		// podSelector.
+		// ExternalEntity or Pod's Namespace matches namespaceSelector and labels matches
+		// podSelector or externalEntitySelector.
 		return true
-	} else if sel.NamespaceSelector != nil {
+	} else if sel.NamespaceSelector != nil && sel.ExternalEntitySelector == nil && sel.PodSelector == nil {
```

Comment:
why need to check for EE and Pod selectors == nil here? if it reaches this else block.. then it is going to be nil no?

Reply:
See the comment above. When the function is called with labelsMatchGroupSelector(pod1, ns1, sel1) and sel1 has both namespaceSelector and externalEntitySelector, we don't want this if condition being hit. w/o checking EE and Pod selectors == nil it will be hit.